### ISSUE TITLE
SSR + NextJS does not seem to Hydrate data

### ISSUE THREAD

**Issue by RickEyre** [tannerlinsley/react-query#1661]
I'm following the guide [here](https://react-query.tanstack.com/guides/ssr) to setup NextJS with Hydration. I can confirm that my dehydrated state is being passed correctly through page props, however on the first SSR render for my page component the hydrated state does not seem to be present. `data` is undefined and my component breaks as I depend on it to be loaded initially. 

```js
export default function App({ Component, pageProps }) {
  // console.log here confirms dehydrated state exists and data has been populated
  return (
    <React.Fragment>
      <QueryClientProvider client={queryClient}>
        <Hydrate state={pageProps.dehydratedState}>
          <Component {...pageProps} />
        </Hydrate>
      </QueryClientProvider>
    </React.Fragment>
  );
}
```

```js
export async function getServerSideProps(context) {
  const { query: { id } } = context;
  const queryClient = new QueryClient();
  
  const api = getAPI(context);
  await api.reAuthenticate();
  
  await queryClient.prefetchQuery(
    [ 'resources', id ],
    () => api.getResource(id)
  );
  
  return {
    props: { dehydratedState: dehydrate(queryClient) }
  };

}
```

```js
export default function Component() {
  const router = useRouter();
  
  // user query client confirms here that dehydrated data is not present
  const { data } = useQuery(
    [ 'resources', router.query.id ],
    () => getAPI().getResource(router.query.id)
  );
  
  // breaks because data is not loaded initially
  return (
    <div>{data.id}</div>
  );
}
```

**Comment by lynxtaa** [comment]
It works in react-query@3.5.13, doesn't work in 3.5.14.
Maybe because of this PR https://github.com/tannerlinsley/react-query/pull/1643 ?

**Comment by TkDodo** [comment]
> Maybe because of this PR #1643 ?

possibly. I was raising that concern in the origin discussion here:
https://github.com/tannerlinsley/react-query/discussions/1641#discussioncomment-279130

**Comment by RickEyre** [comment]
Ah I see. Yes that would break it, I think, as useEffect is not run during SSR in NextJS.

**Comment by tannerlinsley** [comment]
Fixed with eb3d700dc522b86f6e9cc42d08e3b6197766c53b

**Comment by RickEyre** [comment]
Thanks everyone.
